src/runtime_src/hip/core/error.h

```diff
@@ -7,42 +7,166 @@
 #include "core/common/api/bo.h"
 
 #include "hip/config.h"
+#include "hip/core/device.h"
 #include "hip/hip_runtime_api.h"
 
+#include "xrt/experimental/xrt_error.h"
+
 namespace xrt::core::hip
 {
+  /**
+   * @class error
+   * @brief Handles HIP error management, including last error tracking and thread-local error strings.
+   */
   class error
   {
   public:
+    /**
+     * @brief Get the singleton instance of error manager.
+     * @return Reference to the error instance.
+     */
     static error&
     instance();
 
-    static const char*
+    /**
+     * @brief Get error name string for a given HIP error code.
+     * @param err HIP error code.
+     * @return Error name string.
+     */
+    static
+    const char*
     get_error_name(hipError_t err);
 
-    hipError_t
-    peek_last_error()
-    {
-      return m_last_error;
-    }
-
+    /**
+     * @brief Record a thread-local error string for a given HIP error code.
+     * @param err HIP error code.
+     * @param err_str Error string.
+     */
     void
-    reset_last_error()
-    {
-      m_last_error = hipSuccess;
-    }
+    record_local_error(hipError_t err, const std::string& err_str);
 
+    /**
+     * @brief Reset all thread-local error strings.
+     */
     void
-    set_last_error(hipError_t err)
-    {
-      m_last_error = err;
-    }
+    reset_local_errors();
 
-  protected:
-    error();
+    /**
+     * @brief Get the thread-local error string for a given HIP error code.
+     * @param err HIP error code.
+     * @return Error string.
+     */
+    const
+    char*
```

Comment:
Probably a bit easier to read the return type on a single line
```suggestion
    const char*
```